scripts/hide_extra_registration_logs.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+"""Script to hide extra logs that were created by the bug fixed in
+https://github.com/CenterForOpenScience/osf.io/commit/c056262e96be7391eeac7470bfae5426f20bdd88
+
+Finds PROJECT_REGISTERED and REGISTRATION_APPROVAL_APPROVED_LOGS that were added
+due to the bug and sets their should_hide field to True.
+"""
+import sys
+import os
+import json
+import logging
+from website.app import init_app
+from scripts import utils as script_utils
+
+from website import models as mdl
+import datetime as dt
+
+logger = logging.getLogger(__name__)
+
+HERE = os.path.dirname(os.path.abspath(__file__))
+
+# JSON file containing GUIDs of affected registrations
+# Taken from script logs for approve_registrations.py on 2015-10-22
+AFFECTED_REGISTRATION_GUIDS_FILE = os.path.join(HERE, 'hide_extra_registration_logs.json')
+
+# Date when bug was introduced
+# Scripts run at 12am EST == 4:00 UTC
+REGRESSION_DATE = dt.datetime(2015, 10, 8, 4, 0, 0)
+
+# Date fix was put in:
+# https://github.com/CenterForOpenScience/osf.io/commit/c056262e96be7391eeac7470bfae5426f20bdd88
+HOTFIX_DATE = dt.datetime(2015, 10, 22, 4, 30, 0)
+
+def is_errant_log(log):
+    """The approve_registration script saves 2 log types:
+    - PROJECT_REGISTERED
+    - REGISTRATION_APPROVAL_APPROVED
+    """
+    return (
+        # Log already migrated
+        log.should_hide is not True and
+        (
+            log.action in mdl.NodeLog.PROJECT_REGISTERED
+            # The user is None because the
+            # approve_registration script calls
+            # _on_complete with user=None
+            and log.user is None
+        ) or
+        (
+            log.action in mdl.NodeLog.REGISTRATION_APPROVAL_APPROVED
+            # Script is run at 12AM EST or 4:00 UTC
+            # Yes, this is ugly, but there is no other way to distinguish
+            # a log generated by the approve_registration script or a
+            # log generated by a manual approval. Unlike PROJECT_REGISTERED logs,
+            # these will have a user associated with them
+            and log.date.hour == 4
+            and log.date.minute == 0
```

Comment:
Are there any logs where the cron script execution took over a minute, or is its execution always super fast? so its very unlikely?